### PR TITLE
`azurerm_data_factory_flowlet_data_flow`: Optional source and sink properties

### PR DESCRIPTION
**Pull request by ashenm**
Fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/20301

### CODE REVIEW

File: internal/services/datafactory/data_factory_data_flow.go [hunk 1]
```diff
@@ -485,3 +358,142 @@ func flattenDataFactoryDataFlowReference(input *datafactory.DataFlowReference) [
 		},
 	}
 }
+
+func resourceSouceAndSink() *pluginsdk.Resource {
```
Comment (catriona-m) [inline internal/services/datafactory/data_factory_data_flow.go:362]:
```suggestion
func resourceSourceAndSink() *pluginsdk.Resource {
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for submitting this @ashenm - I spotted a typo that I've commented inline but otherwise this is looking good. Thanks!

File: internal/services/datafactory/data_factory_data_flow.go (hunk 1 above)
Comment (ashenm) [reply internal/services/datafactory/data_factory_data_flow.go]: @catriona-m fixed 🙌

**Review by catriona-m — APPROVED** [review]
Thanks @ashenm LGTM!

**Pull request merged**